core/src/test/scala/com/microsoft/azure/synapse/ml/explainers/split1/ICEExplainerSuite.scala

```diff
@@ -0,0 +1,154 @@
+import scala.jdk.CollectionConverters._
+
+
+class ICEExplainerSuite extends TestBase with TransformerFuzzing[ICETransformer] {
+
+  import spark.implicits._
+  val dataDF: DataFrame = (1 to 100).flatMap(_ => Seq(
+    (-5, "a", -5, 0),
+    (-5, "b", -5, 0),
+    (5, "a", 5, 1),
+    (5, "b", 5, 1)
+  )).toDF("col1", "col2", "col3", "label")
+
+  val data: DataFrame = dataDF.withColumn("col4", rand()*100)
+
+  val pipeline: Pipeline = new Pipeline().setStages(Array(
+    new StringIndexer().setInputCol("col2").setOutputCol("col2_ind"),
+    new OneHotEncoder().setInputCol("col2_ind").setOutputCol("col2_enc"),
+    new VectorAssembler().setInputCols(Array("col1", "col2_enc", "col3", "col4")).setOutputCol("features"),
+    new LogisticRegression().setLabelCol("label").setFeaturesCol("features")
+  ))
+  val model: PipelineModel = pipeline.fit(data)
+
+  val ice = new ICETransformer()
+  ice.setModel(model)
+    .setTargetCol("probability")
+    .setCategoricalFeatures(Array(ICECategoricalFeature("col2", Some(2)), ICECategoricalFeature("col3", Some(4))))
+    .setTargetClasses(Array(1))
+  val output: DataFrame = ice.transform(data)
+
+  val iceAvg = new ICETransformer()
+  iceAvg.setModel(model)
+    .setTargetCol("probability")
+    .setCategoricalFeatures(Array(ICECategoricalFeature("col1", Some(100)), ICECategoricalFeature("col2"),
+      ICECategoricalFeature("col3")))
+    .setNumericFeatures(Array(ICENumericFeature("col4", Some(5))))
+    .setTargetClasses(Array(1))
+    .setKind("average")
+  val outputAvg: DataFrame = iceAvg.transform(data)
+
+  test("col2 doesn't contribute to the prediction.") {
+
+    val outputCol2: Map[String, Vector] = outputAvg.select("col2_dependence").collect().map {
+      case Row(map: Map[String, Vector]) =>
+        map
+    }.head
+
+    val impA: Double = outputCol2.get("a").head.toArray.head
+    val impB: Double = outputCol2.get("b").head.toArray.head
+
+    val eps = 0.01
+    assert((impA - impB).abs < eps)
+  }
+
+  test("col3 contribute to the prediction.") {
+
+    val outputCol3: Map[Int, Vector] = outputAvg.select("col3_dependence").collect().map {
+      case Row(map: Map[Int, Vector]) =>
+        map
+    }.head
+
+    val impFirst: Double = outputCol3.get(-5).head.toArray.head
+    val impSec: Double = outputCol3.get(5).head.toArray.head
+
+    assert((impFirst - impSec).abs > 0.4)
+  }
+
+  test("The length of explainer map for numeric feature is equal to it's numSplits.") {
+
+    val outputCol1: Map[Double, Vector] = outputAvg.select("col4_dependence").collect().map {
+      case Row(map: Map[Double, Vector]) =>
+        map
+    }.head
+
+    assert(outputCol1.size == iceAvg.getNumericFeatures.head.getNumSplits + 1)
+  }
+
+  test("The length of explainer map for categorical feature is less or equal to it's numTopValues.") {
+    val outputCol: Map[Double, Vector] = output.select("col3_dependence").collect().map {
+      case Row(map: Map[Double, Vector]) =>
+        map
+    }.head
```

Comment:
this code is repeated in a few places, consider pulling into a function to keep it dry